### ISSUE TITLE
Any plans to add the unsloth GRPO implementation?

### ISSUE THREAD

**Issue by Vattikondadheeraj** [pytorch/torchtune#2417]
Hey, I have seen that the GRPO is under the hoods and I have played with the code a bit. I was wondering if there are any plans to add the unsloth implementation of GRPO? Maybe the adding KL divergence and modifying the loss function

[https://unsloth.ai/blog/grpo](url)

**Comment by ebsmothers** [comment]
Hi @Vattikondadheeraj thanks for creating the issue. Can you clarify your comment about the KL divergence? I took a look at both #2326's `GRPOSimpleLoss` and unsloth's `grpo_compute_loss` and I don't immediately see a difference. E.g. [this](https://github.com/RedTachyon/torchtune/blob/main/torchtune/dev/grpo/loss.py#L243-L247) and [this](https://github.com/unslothai/unsloth-zoo/blob/a9857088bdaf412bef36800d837a3a37657555c8/unsloth_zoo/rl_replacements.py#L64) look similar to me (but would need to do a more detailed comparison to say for sure). The linear cross-entropy trick looks like a version of our chunked cross entropy (via an autograd function) with some extra bells and whistles. I think as a first step we can look at [Horace's chunked linear cross-entropy](https://gist.github.com/Chillee/22cd93e11b887db1f596ab754d60a899) for our own CE loss then try to extend to GRPO
